api/src/org/labkey/api/wiki/WikiRenderingService.java

```diff
@@ -15,6 +15,14 @@ public interface WikiRenderingService
     HtmlString WIKI_PREFIX = HtmlString.unsafe("<div class=\"labkey-wiki\">");
     HtmlString WIKI_SUFFIX = HtmlString.unsafe("</div>");
 
+    // How should substitutions be handled? Substitutions are allowed in HTML wikis only.
+    enum SubstitutionMode
+    {
+        Ignore,
+        Remove,
+        Substitute
```

Comment:
It took me a minute to figure out the behavior. The names are fine, but check if these comments are accurate and commit them if you think they're helpful.

```suggestion
        Ignore, // Don't substitute, leaving any substitution tokens as-is
        Remove, // Substitute with blanks, ensuring that all substitution tokens are removed
        Substitute // Substitute normally, swapping substitution tokens with their replacements
```